### PR TITLE
Consistent `Handler` naming

### PR DESCRIPTION
**Pull request by nnethercote**
This PR implements the renaming described in https://github.com/rust-lang/compiler-team/issues/699.

r? @compiler-errors

### CODE REVIEW

**Comment by nnethercote** [comment]
Apologies for the massively tedious review. I split it into lots of commits so that each commit (except the last few) does exactly one change. And I already have conflicts since I started this yesterday, lol.

File: compiler/rustc_codegen_cranelift/src/base.rs [hunk 1]
```diff
@@ -176,10 +176,10 @@ pub(crate) fn compile_fn(
         match module.define_function(codegened_func.func_id, context) {
             Ok(()) => {}
             Err(ModuleError::Compilation(CodegenError::ImplLimitExceeded)) => {
-                let handler = rustc_session::EarlyErrorHandler::new(
+                let early_dcx = rustc_session::EarlyDiagCtxt::new(
```
Comment (compiler-errors) [inline compiler/rustc_codegen_cranelift/src/base.rs:179]: Unrelated, but why are codegen backends using their own diagnostic contexts 🤔

File: compiler/rustc_query_system/src/dep_graph/graph.rs [hunk 2]
```diff
@@ -924,7 +924,7 @@ impl<D: Deps> DepGraphData<D> {
             // Promote the previous diagnostics to the current session.
             qcx.store_side_effects(dep_node_index, side_effects.clone());
 
-            let handle = qcx.dep_context().sess().diagnostic();
+            let handle = qcx.dep_context().sess().dcx();
```
Comment (compiler-errors) [inline compiler/rustc_query_system/src/dep_graph/graph.rs:927]:
```suggestion
            let dcx = qcx.dep_context().sess().dcx();
```

I assume `handle` was meant to be `handler`.

File: compiler/rustc_codegen_llvm/src/back/lto.rs [hunk 3]
```diff
@@ -200,18 +200,11 @@ pub(crate) fn run_fat(
     modules: Vec<FatLtoInput<LlvmCodegenBackend>>,
     cached_modules: Vec<(SerializedModule<ModuleBuffer>, WorkProduct)>,
 ) -> Result<LtoModuleCodegen<LlvmCodegenBackend>, FatalError> {
-    let diag_handler = cgcx.create_diag_handler();
-    let (symbols_below_threshold, upstream_modules) = prepare_lto(cgcx, &diag_handler)?;
+    let dcx = cgcx.create_dcx();
```
Comment (compiler-errors) [inline compiler/rustc_codegen_llvm/src/back/lto.rs:203]: I wonder if this should be `create_diag_ctxt` or something

**Review by compiler-errors — APPROVED** [review]
r=me after responding to some questions + one nit

File: compiler/rustc_codegen_cranelift/src/base.rs (hunk 1 above)
Comment (nnethercote) [reply compiler/rustc_codegen_cranelift/src/base.rs:179]: Good question. I've added looking into this to my errors TODO list.

File: compiler/rustc_query_system/src/dep_graph/graph.rs (hunk 2 above)
Comment (nnethercote) [reply compiler/rustc_query_system/src/dep_graph/graph.rs]: Good catch. I did lots of searching with regexps to find as many of the old names as I could, but I didn't look for `handle`. Looks like this was the only place it was used for a `Handler`, and presumably it was a typo. I have updated the code to change this one.

File: compiler/rustc_codegen_llvm/src/back/lto.rs (hunk 3 above)
Comment (nnethercote) [reply compiler/rustc_codegen_llvm/src/back/lto.rs:203]: I don't see why function names need to be different to variable names. Consistency is good.

Comment (compiler-errors) [reply compiler/rustc_codegen_llvm/src/back/lto.rs:203]: It's consistent with variable names, but not the type name. I consider it more normal for a constructor function to be called create_type_name, personally.

**Comment by nnethercote** [comment]
Thank you for the fast review!

@bors r=compiler-errors

**Comment by nnethercote** [comment]
I fixed the rustfmt failure.

@bors r=compiler-errors p=1

High-priority because it's conflict prone.

File: compiler/rustc_codegen_cranelift/src/base.rs (hunk 1 above)
Comment (nnethercote) [reply compiler/rustc_codegen_cranelift/src/base.rs:179]: This function has a `CodegenCx`, which is constructed from a `tcx` but does not hold on to the `tcx`. So a `dcx` ref could be added to `CodegenCx` pretty easily, and then these diagnostic contexts could be removed.

**Pull request merged**